pkg/cluster/internal/create/create.go

```diff
@@ -83,14 +76,6 @@ func Cluster(logger log.Logger, p providers.Provider, opts *ClusterOptions) erro
 		return err
 	}
 
-	// TODO: move to config validation
-	// validate the name
-	if !validNameRE.MatchString(opts.Config.Name) {
-		return errors.Errorf(
-			"'%s' is not a valid cluster name, cluster names must match `%s`",
-			opts.Config.Name, validNameRE.String(),
-		)
-	}
 	// warn if cluster name might typically be too long
 	if len(opts.Config.Name) > clusterNameMax {
```

Comment:
should this warning be moved to the validation as well?

Reply:
maybe, I'm not sure if it makes sense for Validate() to warn, and this is not inherent in the API (we don't enforce this restriction ourselves), it's a limitation of the underlying implementation currently